### PR TITLE
Ncnews4

### PR DESCRIPTION
**Pull request by Nat2llaw**
added happy and sad path with tdd for tasks 4

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,18 +36,9 @@ describe("GET/api/articles/:article_id", () => {
...
-            title: "Living in the shadow of a great man",
-            topic: "mitch",
-            author: "butter_bridge",
-            body: "I find this existence challenging",
-            created_at: "2020-07-09T20:11:00.000Z",
-            votes: 100,
-          },
-        ]);
+        .then(({ body }) => {
+          console.log(body)
```
Comment (MRombi) [inline __tests__/app.test.js:48]: You should remove console.log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -26,28 +26,26 @@ describe("GET/api/topics", () => {
...
       .expect(404)
       .then(({ body }) => {
-        expect.objectContaining({});
+        expect(body).toEqual({});
       });
   });
 });
 
-// describe("GET/api/articles/:article_id", () => {
-//   test("200: return article by id", () => {
```
Comment (MRombi) [inline __tests__/app.test.js:35]: Hi Nathaniel, this test was good as it checks if you are getting the correct article with your request.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -26,28 +26,26 @@ describe("GET/api/topics", () => {
...
+describe("GET/api/articles/:article_id", () => {
+  test("200: return article by id", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect.objectContaining(body);
+      });
+  });
+  test("404:article id not in database", () => {
```
Comment (MRombi) [inline __tests__/app.test.js:43]: The test should be more specific and check whether you are receiving the actual error message you are sending.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -26,28 +26,26 @@ describe("GET/api/topics", () => {
...
+    return request(app)
+      .get("/api/articles/1123")
+      .expect(404)
+        .then(({ body }) => {
+          expect(body).toEqual({});
+      });
```
Comment (corkesw) [inline __tests__/app.test.js:49]: I would not expect an empty object here, I'd expect a key of message (or msg) with some meaningful feedback as the value (eg, "article not found")

**Review by corkesw — COMMENTED** [review]
We generally send things back from the api in an object on a relevant key. Eg an article would come back in an object on a key of article, a list of topics would come back in an object on a key of topics. Keeping this consistent will allow us to extend our api at a later date and will make things easier when we get to the front end.

**Pull request merged**